crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -0,0 +1,327 @@
+use ruff_diagnostics::Diagnostic;
+use ruff_diagnostics::Violation;
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::visitor::{self, Visitor};
+use ruff_python_ast::{self as ast, Expr, Stmt};
+use ruff_python_semantic::SemanticModel;
+use ruff_python_semantic::{Definition, MemberKind};
+use ruff_text_size::TextRange;
+
+use crate::checkers::ast::Checker;
+use crate::docstrings::sections::{SectionContexts, SectionKind};
+use crate::docstrings::styles::SectionStyle;
+use crate::registry::Rule;
+use crate::rules::pydocstyle::settings::Convention;
+
+/// ## What it does
+/// Checks for function docstrings that do not include documentation for all
+/// raised exceptions.
+///
+/// ## Why is this bad?
+/// This rule helps prevent you from leaving docstrings unfinished or incomplete.
+/// Some conventions require all explicit exceptions to be documented.
+///
+/// ## Example
+/// ```python
+/// def calculate_speed(distance: float, time: float) -> float:
+///     """Calculate speed as distance divided by time.
+///
+///     Args:
+///         distance: Distance traveled.
+///         time: Time spent traveling.
+///
+///     Returns:
+///         Speed as distance divided by time.
+///     """
+///     try:
+///         return distance / time
+///     except ZeroDivisionError as exc:
+///         raise FasterThanLightError from exc
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def calculate_speed(distance: float, time: float) -> float:
+///     """Calculate speed as distance divided by time.
+///
+///     Args:
+///         distance: Distance traveled.
+///         time: Time spent traveling.
+///
+///     Returns:
+///         Speed as distance divided by time.
+///
+///     Raises:
+///         FasterThanLightError: If speed is greater than the speed of light.
+///     """
+///     try:
+///         return distance / time
+///     except ZeroDivisionError as exc:
+///         raise FasterThanLightError from exc
```

Comment:
What if this is like, `raise foo.FasterThanLightError from exc`? Would / should the docstring contain `FasterThanLightError` or `foo.FasterThanLightError`?

Reply:
Hard to say. I found nothing in the google style guide or numpy style guide. Per your other comments, I'll try to make it agnostic.